### DISCUSSION TITLE
How dde.grad.jacobian compute gradients in PDE

### DISCUSSION THREAD

**Discussion by dang-trinh156** [lululxvi/deepxde#1247] [Q&A] [answered]
Dear, Prof. Lululxvi and DeepXDE users,

I am viewing Schrodinger.ipynb example, and I would like to modify PDE equation from the original code:
`def pde(x, y):
    """
    INPUTS:
        x: x[:,0] is x-coordinate
           x[:,1] is t-coordinate
        y: Network output, in this case:
            y[:,0] is u(x,t) the real part
            y[:,1] is v(x,t) the imaginary part
    OUTPUT:
        The pde in standard form i.e. something that must be zero
    """

    u = y[:, 0:1]
    v = y[:, 1:2]

    # In 'jacobian', i is the output component and j is the input component
    u_t = dde.grad.jacobian(y, x, i=0, j=1)
    v_t = dde.grad.jacobian(y, x, i=1, j=1)

    u_x = dde.grad.jacobian(y, x, i=0, j=0)
    v_x = dde.grad.jacobian(y, x, i=1, j=0)

    # In 'hessian', i and j are both input components. (The Hessian could be in principle something like d^2y/dxdt, d^2y/d^2x etc)
    # The output component is selected by "component"
    u_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)
    v_xx = dde.grad.hessian(y, x, component=1, i=0, j=0)

    f_u = u_t + 0.5 * v_xx + (u ** 2 + v ** 2) * v
    f_v = v_t - 0.5 * u_xx - (u ** 2 + v ** 2) * u

    return [f_u, f_v]`
And would changing 
`u_t = dde.grad.jacobian(y, x, i=0, j=1)`
`v_t = dde.grad.jacobian(y, x, i=1, j=1)`
`u_x = dde.grad.jacobian(y, x, i=0, j=0)`
 `v_x = dde.grad.jacobian(y, x, i=1, j=0)`
to
`u_t = dde.grad.jacobian(u, x, j=1)`
`v_t = dde.grad.jacobian(v, x, j=1)`
`u_x = dde.grad.jacobian(u, x, j=0)`
`v_x = dde.grad.jacobian(v, x, j=0)`
have any significant impacts on the performance of PINN, or it's totally different from the original way to compute gradients of dde.grad.jacobian?
P/s: I have tested and  it still runs normally, but I wouldn't figure out why as the weights and losses can't be controlled evert time I run the code.

**Comment by lululxvi** [comment, marked as answer]
`dde.grad.jacobian(y, x, i=0, j=1)` and `dde.grad.jacobian(u, x, j=1)` are not equal, see https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant and https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#deepxde.gradients.jacobian

**Reply by dang-trinh156** [reply]
Thank you for your reply!
